x-pack/plugins/observability/server/services/slo/tasks/orphan_summary_cleanup_task.ts

```diff
@@ -0,0 +1,246 @@
+              },
+            },
+          });
+        }
+      } while (searchAfterKey);
+    }
+
+    return { state: { lastRunAt: runAt } };
+  };
+
+  fetchSloSummariesIds = async (
+    searchAfter?: AggregationsCompositeAggregateKey
+  ): Promise<{
+    searchAfter?: AggregationsCompositeAggregateKey;
+    sloSummaryIds: Array<{ id: string; revision: number }>;
+  }> => {
+    this.logger.debug(`[SLO] Fetching SLO Summaries ids after ${searchAfter}`);
+    if (!this.esClient) {
+      return {
+        searchAfter: undefined,
+        sloSummaryIds: [],
+      };
+    }
+
+    const size = 1000;
+
+    const result = await this.esClient.search<
+      unknown,
+      {
+        slos: {
+          after_key: AggregationsCompositeAggregateKey;
+          buckets: Array<{
+            key: {
+              id: string;
+              revision: number;
+            };
+          }>;
+        };
+      }
+    >({
+      size: 0,
+      index: SLO_SUMMARY_DESTINATION_INDEX_PATTERN,
+      aggs: {
+        slos: {
+          composite: {
+            size,
+            sources: [
+              {
+                id: {
+                  terms: {
+                    field: 'slo.id',
+                  },
+                },
+              },
+              {
+                revision: {
+                  terms: {
+                    field: 'slo.revision',
+                  },
+                },
+              },
+            ],
+            after: searchAfter,
+          },
+        },
+      },
+    });
+
+    const aggBuckets = result.aggregations?.slos.buckets ?? [];
+    if (aggBuckets.length === 0) {
+      return {
+        searchAfter: undefined,
+        sloSummaryIds: [],
+      };
+    }
+
+    const newSearchAfter =
+      aggBuckets.length < size ? undefined : result.aggregations?.slos.after_key;
+
+    const sloSummaryIds = aggBuckets.map(({ key }) => {
+      return {
+        id: key.id as string,
+        revision: key.revision as number,
+      };
+    });
+
+    return {
+      searchAfter: newSearchAfter,
+      sloSummaryIds,
+    };
+  };
+
+  private get taskId() {
+    return `${TASK_TYPE}:1.0.0`;
+  }
+
+  public async start(
+    taskManager: TaskManagerStartContract,
+    soClient: SavedObjectsClientContract,
+    esClient: ElasticsearchClient
+  ) {
+    this.taskManager = taskManager;
+    this.soClient = soClient;
+    this.esClient = esClient;
+
+    if (!taskManager) {
+      this.logger.info('[SLO] Missing required service during startup, skipping task.');
+    }
+
+    if (this.config.sloOrphanSummaryCleanUpTaskEnabled) {
+      this.taskManager.ensureScheduled({
+        id: this.taskId,
+        taskType: TASK_TYPE,
+        schedule: {
+          interval: '1h',
+        },
+        scope: ['observability', 'slo'],
+        state: {},
+        params: {},
+      });
+    } else {
+      this.taskManager.removeIfExists(this.taskId);
+    }
+  }
```

Comment:
I'm trying to understand if the taskId can exist while the config is false? Is it possible? 

Or is a taskId persisted between kibana restart? And therefore if we start kibana with the config=true, then change config=false and restart it, we do need to remove the task?

Reply:
if it's false we are removing it as you can see in the else condition 

`this.taskManager.removeIfExists(this.taskId);`